tests/test_init_signature.py

```diff
@@ -0,0 +1,125 @@
+from inspect import Parameter, signature
+from typing import Any
+
+import pytest
+
+from pydantic import BaseModel, Extra, compiled
+
+pos_or_kw = Parameter.POSITIONAL_OR_KEYWORD
+kw_only = Parameter.KEYWORD_ONLY
+empty = Parameter.empty
+
+
+def test_init_signature():
+    class Model(BaseModel):
+        a: float
+        b: int = 10
+
+    assert BaseModel.__init__ is not Model.__init__
+    sig = signature(Model.__init__)
+    assert sig != signature(BaseModel.__init__)
+
+    params = sig.parameters
```

Comment:
I think this would be much easier to read and debug with something like 

```py
assert {n: str(p) for n, p in s.parameters.items()} = {
    ...
}
```

Reply:
Yeah, thats' right. But I'd went with
```py
assert [str(p) for p in sig.parameters.values()] == [
        'self',
        'id: int = 1',
        'bar=2',
        'baz: Any',
        "name: str = 'John Doe'",
        '**data',
    ]
```
to maintain the order.

Reply:
Okay

Reply:
Done!